crypto/cryptlib.c

```diff
@@ -396,15 +396,34 @@ void OPENSSL_showfatal(const char *fmta, ...)
 # endif
 }
 #else
+
+# ifdef __APPLE__
+#  include <Availability.h>
+#   if (defined(__MAC_10_11)  && __MAC_OS_X_VERSION_MIN_REQUIRED  >= __MAC_10_11) || \
+       (defined(__IPHONE_9_0) && __IPHONE_OS_VERSION_MIN_REQUIRED >= __IPHONE_9_0)
+#    define HAVE_OS_CRASH
+extern void _os_crash(const char *);
+#   endif
+# endif
+
 void OPENSSL_showfatal(const char *fmta, ...)
 {
-#ifndef OPENSSL_NO_STDIO
     va_list ap;
 
     va_start(ap, fmta);
+
+#ifdef HAVE_OS_CRASH
+    va_list ap2;
+    va_copy(ap2, ap);
```

Comment:
We don't allow mixed declarations and code. Declarations should always be at the beginning of the block.